apps/web/src/components/Basenames/RenewalForm/index.tsx

```diff
@@ -0,0 +1,186 @@
+    [basenameChain.id, switchChain],
+  );
+
+  const isOnSupportedNetwork = useMemo(
+    () => connectedChain && supportedChainIds.includes(connectedChain.id),
+    [connectedChain],
+  );
+
+  const nameData = useMemo(() => {
+    return namesData?.data.find((n) => n.domain === name);
+  }, [name, namesData?.data]);
+
+  const formattedExpirationDate = useMemo(() => {
+    if (!nameData?.expires_at) return undefined;
+    const date = new Date(nameData.expires_at);
+    return date.toLocaleDateString('en-US', {
+      month: '2-digit',
+      day: '2-digit',
+      year: 'numeric',
+    });
+  }, [nameData?.expires_at]);
+
+  const {
+    callback: renewBasename,
+    value: price,
+    isPending,
+    renewNameStatus,
+    batchCallsStatus,
+  } = useRenewNameCallback({
+    name,
+    years,
+  });
+
+  const increment = useCallback(() => {
+    setYears((n) => n + 1);
+  }, [setYears]);
+
+  const decrement = useCallback(() => {
+    setYears((n) => (n > 1 ? n - 1 : n));
+  }, [setYears]);
+
+  const ethUsdPrice = useEthPriceFromUniswap();
+
+  const renewName = useCallback(async () => {
+    try {
+      logEventWithContext('renew_name_initiated', ActionType.click);
+      await renewBasename();
+    } catch (error) {
+      logError(error, 'Failed to renew name');
+    }
+  }, [logEventWithContext, logError, renewBasename]);
+
+  const renewNameCallback = useCallback(() => {
+    renewName().catch((e) => {
+      logError(e, 'Failed to renew name');
+    });
+  }, [logError, renewName]);
+
+  const { auxiliaryFunds: auxiliaryFundsEnabled } = useCapabilitiesSafe({
+    chainId: connectedChain?.id,
+  });
+  const { data: balance } = useBalance({ address, chainId: connectedChain?.id });
+  const insufficientBalanceToRenew =
+    balance?.value !== undefined && price !== undefined && balance?.value < price;
+  const correctChain = connectedChain?.id === basenameChain.id;
+  const insufficientFundsAndNoAuxFunds = insufficientBalanceToRenew && !auxiliaryFundsEnabled;
+  const insufficientBalanceToRenewAndCorrectChain = insufficientBalanceToRenew && correctChain;
+  const insufficientFundsNoAuxFundsAndCorrectChain =
+    !auxiliaryFundsEnabled && insufficientBalanceToRenewAndCorrectChain;
+
+  const hasResolvedUSDPrice = price !== undefined && ethUsdPrice !== undefined;
+  const usdPrice = hasResolvedUSDPrice ? formatUsdPrice(price, ethUsdPrice) : '--.--';
+
+  const mainRenewalElementClasses = classNames(
+    'z-10 flex flex-col items-start justify-between gap-6 bg-[#F7F7F7] p-8 text-gray-60 shadow-xl md:flex-row md:items-center relative z-20 rounded-2xl',
+  );
+
+  useEffect(() => {
+    if (
+      renewNameStatus === WriteTransactionWithReceiptStatus.Success ||
+      batchCallsStatus === BatchCallsStatus.Success
+    ) {
+      setYears(1);
+      refetch().catch((e) => {
+        logError(e, 'Failed to refetch names');
+      });
+    }
+  }, [renewNameStatus, batchCallsStatus, refetch, logError]);
+
+  if (!isOnSupportedNetwork) {
```

Comment:
```suggestion
  if (address && !isOnSupportedNetwork) {
```

would only show this if the user is connected, otherwise it's a confusing CTA and `switchToIntendedNetwork` doesn't do anything. 

![image](https://github.com/user-attachments/assets/b7aa8f4b-7ae2-4f20-980b-974067608e3d)